### DISCUSSION TITLE
Jobs failing with uncaught-error status: .git/index.lock: No space left on device

### DISCUSSION THREAD

**Discussion by rdsubhas** [renovatebot/renovate#9259] [Request Help]
**What Renovate type, platform and version are you using?**

* Type: **Renovate Hosted App**
* Platform: **GitHub**
* Language: **javascript**
* Manager: **yarn**

**Describe the bug**

Renovate jobs on a private repo continuously failing with `caught-error` status in dashboard. Root cause in the logs: `Unable to create '/mnt/renovate/gh/goeuro/apps-integrator/.git/index.lock': No space left on device\n"`

<img width="348" alt="image" src="https://user-images.githubusercontent.com/1570745/112199716-b9d35e00-8c0e-11eb-882f-84e6636aec40.png">

**Relevant debug logs**

Job # 322633804

```
DEBUG: lock file error
DEBUG: Unknown error committing files
...
Unable to create '/mnt/renovate/gh/goeuro/apps-integrator/.git/index.lock': No space left on device\n"

ERROR: Disk space error - skipping
```

**Have you created a minimal reproduction repository?**

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

-----
cc @vladkosinov

**Comment by rarkins** [comment]
How big is the repo when cloned and all modules installed?

**Reply by vladkosinov** [reply]
```bash
du -sh
2.4G
```

**Reply by rarkins** [reply]
We use a 4GB SSD by default so normally don't have any problems unless the size of the repo (including .git) and all modules exceeds 3 or more GB.

I'll switch this repo to EBS storage which allows for ~10GB and should have enough headroom.

**Comment by brettbatie** [comment]
@rarkins  I'm encountering the same issue. job # #490367666. It keeps trying every couple of hours but continues to fail.

With a `git clone` the mono repo is 1.2 GB and after a `yarn install` it is 3.8 GB. Any recommendations?

---
```
DEBUG: PR has no releaseTimestamp(branch="renovate/pin-dependencies")
DEBUG: 8 file(s) to commit(branch="renovate/pin-dependencies")
DEBUG: Committing files to branch renovate/pin-dependencies(branch="renovate/pin-dependencies")
DEBUG: Setting git author name(branch="renovate/pin-dependencies")
{
  "gitAuthorName": "Renovate Bot"
}
DEBUG: Setting git author email(branch="renovate/pin-dependencies")
{
  "gitAuthorEmail": "bot@renovateapp.com"
}
DEBUG: Unknown error committing files(branch="renovate/pin-dependencies")
{
  "err": {
    "task": {
      "commands": [
        "reset",
        "--hard"
      ],
      "format": "utf-8"
    },
    "message": "error: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/mnt/renovate/gh/brettbatie/cleandrive/.git/refs/heads/master.lock': No space left on device\n",
    "stack": "Error: error: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/mnt/renovate/gh/brettbatie/cleandrive/.git/refs/heads/master.lock': No space left on device\n\n    at Object.action (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/error-detection.plugin.js:30:33)\n    at PluginStore.exec (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/plugin-store.js:21:33)\n    at /home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:95:45\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:93:16)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:77:46)\n    at Generator.next (<anonymous>)\n    at fulfilled (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
  }
}
ERROR: Disk space error - skipping
```

**Reply by rarkins** [reply]
I've moved your repo onto a larger partition and jobs are now completing
